openmc/mgxs/mdgxs.py

```diff
@@ -605,20 +677,45 @@ def print_xs(self, subdomains='all', nuclides='all', xs_type='macro'):
 
                     template = '{0: <12}Group {1} [{2: <10} - {3: <10}eV]:\t'
 
-                    # Loop over energy groups ranges
-                    for group in range(1, self.num_groups+1):
-                        bounds = self.energy_groups.get_group_bounds(group)
-                        string += template.format('', group, bounds[0], bounds[1])
-                        average = self.get_xs([group], [subdomain], [nuclide],
-                                              xs_type=xs_type, value='mean',
-                                              delayed_groups=[delayed_group])
-                        rel_err = self.get_xs([group], [subdomain], [nuclide],
-                                              xs_type=xs_type, value='rel_err',
-                                              delayed_groups=[delayed_group])
-                        average = average.flatten()[0]
-                        rel_err = rel_err.flatten()[0] * 100.
-                        string += '{:.2e} +/- {:1.2e}%'.format(average, rel_err)
-                        string += '\n'
+                    average_xs = self.get_xs(nuclide=[nuclide],
+                                             subdomain=[subdomain],
+                                             xs_type=xs_type, value='mean',
+                                             delayed_groups=[delayed_group])
+                    rel_err_xs = self.get_xs(nuclide=[nuclide],
+                                             subdomain=[subdomain],
+                                             xs_type=xs_type, value='rel_err',
+                                             delayed_groups=[delayed_group])
+                    rel_err_xs = rel_err_xs * 100.
+
+                    if self.num_polar or self.num_azimuthal:
```

Comment:
If we make `num_polar` and `num_azimuthal` default to unity, I think it would simplify this `if-else` block and you can embed  the loop over angles directly into a single nested `for` loop block.